plugins/outputs/sql/sql_test.go

```diff
@@ -434,3 +437,87 @@ func TestClickHouseDsnConvert(t *testing.T) {
 		require.Equal(t, tt.expected, plugin.DataSourceName)
 	}
 }
+
+func TestMysqlEmptyTimestampColumnIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	initdb, err := filepath.Abs("testdata/mariadb_no_timestamp/initdb/script.sql")
+	require.NoError(t, err)
+
+	// initdb/script.sql creates this database
+	const dbname = "foo"
+
+	// The mariadb image lets you set the root password through an env
+	// var. We'll use root to insert and query test data.
+	const username = "root"
+
+	password := testutil.GetRandomString(32)
+	outDir := t.TempDir()
+
+	servicePort := "3306"
+	container := testutil.Container{
+		Image: "mariadb",
+		Env: map[string]string{
+			"MARIADB_ROOT_PASSWORD": password,
+		},
+		Files: map[string]string{
+			"/docker-entrypoint-initdb.d/script.sql": initdb,
+			"/out":                                   outDir,
+		},
+		ExposedPorts: []string{servicePort},
+		WaitingFor: wait.ForAll(
+			wait.ForListeningPort(nat.Port(servicePort)),
+			wait.ForLog("mariadbd: ready for connections.").WithOccurrence(2),
+		),
+	}
+	require.NoError(t, container.Start(), "failed to start container")
+	defer container.Terminate()
```

Comment:
The container termination is deferred, but if an error occurs early in the setup, the container might not be terminated. Consider a `t.Cleanup()` function to ensure proper cleanup even if `require.NoError(t, container.Start())` fails.

Reply:
Re-read the code, `require.NoError(t, container.Start())` starts the container, followed by `defer container.Terminate()`. However, if an error occurs before reaching `defer` (e.g., during file path resolution), the container won’t be terminated properly.  

Using `t.Cleanup()` instead ensures cleanup runs even if the test fails early.  

Please see the suggestion below. let me know your thoughts!